src/box/iproto_constants.h

```diff
@@ -49,255 +57,270 @@ enum {
 	XLOG_FIXHEADER_SIZE = 19
 };
 
-/**
- * IPROTO_FLAGS bitfield constants.
- * `box.iproto.flag` needs to be updated correspondingly.
- */
-enum {
-	/** Set for the last xrow in a transaction. */
-	IPROTO_FLAG_COMMIT = 0x01,
-	/** Set for the last row of a tx residing in limbo. */
-	IPROTO_FLAG_WAIT_SYNC = 0x02,
-	/** Set for the last row of a synchronous tx. */
-	IPROTO_FLAG_WAIT_ACK = 0x04,
-};
-
-/**
- * `box.iproto.key` needs to be updated correspondingly.
- */
-enum iproto_key {
-	IPROTO_REQUEST_TYPE = 0x00,
-	IPROTO_SYNC = 0x01,
-
-	/* Replication keys (header) */
-	IPROTO_REPLICA_ID = 0x02,
-	IPROTO_LSN = 0x03,
-	IPROTO_TIMESTAMP = 0x04,
-	IPROTO_SCHEMA_VERSION = 0x05,
-	IPROTO_SERVER_VERSION = 0x06,
-	IPROTO_GROUP_ID = 0x07,
-	IPROTO_TSN = 0x08,
-	IPROTO_FLAGS = 0x09,
-	IPROTO_STREAM_ID = 0x0a,
-	/* Leave a gap for other keys in the header. */
-	IPROTO_SPACE_ID = 0x10,
-	IPROTO_INDEX_ID = 0x11,
-	IPROTO_LIMIT = 0x12,
-	IPROTO_OFFSET = 0x13,
-	IPROTO_ITERATOR = 0x14,
-	IPROTO_INDEX_BASE = 0x15,
-	/* Leave a gap between integer values and other keys */
+/** IPROTO_FLAGS bitfield constants. */
+#define IPROTO_FLAGS(_)						\
+	/** Set for the last xrow in a transaction. */		\
+	_(IPROTO_FLAG_COMMIT, 0x01)				\
+	/** Set for the last row of a tx residing in limbo. */	\
+	_(IPROTO_FLAG_WAIT_SYNC, 0x02)				\
+	/** Set for the last row of a synchronous tx. */	\
+	_(IPROTO_FLAG_WAIT_ACK, 0x04)				\
+
+ENUM(iproto_flag, IPROTO_FLAGS);
+
+/** Constants generated from IPROTO_FLAGS. */
+extern struct iproto_constant iproto_flag_constants[];
+
+/** Size of iproto_flag_constants. */
+extern size_t iproto_flag_constants_size;
+
+#define IPROTO_KEYS(_)				\
+	_(IPROTO_REQUEST_TYPE, 0x00)		\
+	_(IPROTO_SYNC, 0x01)			\
+						\
+	/* Replication keys (header) */		\
+	_(IPROTO_REPLICA_ID, 0x02)		\
+	_(IPROTO_LSN, 0x03)			\
+	_(IPROTO_TIMESTAMP, 0x04)		\
+	_(IPROTO_SCHEMA_VERSION, 0x05)		\
+	_(IPROTO_SERVER_VERSION, 0x06)		\
+	_(IPROTO_GROUP_ID, 0x07)		\
+	_(IPROTO_TSN, 0x08)			\
+	_(IPROTO_FLAGS, 0x09)			\
+	_(IPROTO_STREAM_ID, 0x0a)		\
+	/* Leave a gap for other keys in the header. */	\
+	_(IPROTO_SPACE_ID, 0x10)		\
+	_(IPROTO_INDEX_ID, 0x11)		\
+	_(IPROTO_LIMIT, 0x12)			\
+	_(IPROTO_OFFSET, 0x13)			\
+	_(IPROTO_ITERATOR, 0x14)		\
+	_(IPROTO_INDEX_BASE, 0x15)		\
+	/* Leave a gap between integer values and other keys */	\
 	/**
 	 * Flag indicating the need to send position of
 	 * last selected tuple in response.
```

Comment:
And what's bad about it?

It would like this:

```C
_(IPROTO_INDEX_BASE, 0x15)		\
_(IPROTO_FETCH_POSITION, 0x1f, "Flag indicating the need to send position of last selected tuple in response.")		\
```

Reply:
Got it.
That's a good idea, but some comments are too long to be turned into one-liner (or contain schemes), and we will need to put line continuation tokens at every line of comment - try to delete line continuation token from [here](https://godbolt.org/z/aYzffPK57).